addons/binding/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/netutils/WebSocketConnection.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.deconz.internal.netutils;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketClose;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketConnect;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketError;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketMessage;
+import org.eclipse.jetty.websocket.api.annotations.WebSocket;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.openhab.binding.deconz.internal.dto.SensorMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * Establishes and keeps a websocket connection to the deCONZ software.
+ *
+ * The connection is closed by deCONZ now and then and needs to be re-established.
+ *
+ * @author David Graeff - Initial contribution
+ */
+@WebSocket
+@NonNullByDefault
+public class WebSocketConnection {
+    private final Logger logger = LoggerFactory.getLogger(WebSocketConnection.class);
+
+    private WebSocketClient client = new WebSocketClient();
```

Comment:
Do not instantiate it yourself! Use the shared client that is available from ESH's `WebSocketFactory`.